sdk/spring/azure-spring-integration-servicebus/src/test/java/com/azure/spring/integration/servicebus/converter/ServiceBusMessageConverterTest.java

```diff
@@ -205,20 +232,42 @@ public void testServiceBusMessageHeadersSet() {
 
         String customHeader = "custom-header";
         String customHeaderValue = "custom-header-value";
+        // when session id set, the partition key equals to session id.
+        Message<String> springMessage = springMessageBuilder().setHeader(PARTITION_KEY, SERVICE_BUS_SESSION_ID)
```

Comment:
We can set PARTITION_KEY and AzureHeaders.PARTITION_KEY with different values from SESSION_ID, and test the if converted header is as expected.